src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -56,12 +56,12 @@ private BoundExpression LowerDeclarationPattern(BoundDeclarationPattern pattern,
             if (pattern.IsVar)
             {
                 Debug.Assert(input.Type == pattern.LocalSymbol.Type);
-                var assignment = _factory.AssignmentExpression(_factory.Local(pattern.LocalSymbol), input);
+                var assignment = _factory.AssignmentExpression(pattern.VariableAccess, input);
```

Comment:
Currently, it is maintained in fully lowered form at all times. It wouldn't hurt, though.

---
In reply to: [77386791](https://github.com/dotnet/roslyn/pull/13535#discussion_r77386791) [](ancestors = 77386791)